plugins/auth-backend/src/lib/passport/BackstageIdentityStrategy.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Request } from 'express';
+import { JWK, JWT, JWKS } from 'jose';
+import { BackstageIdentity } from '../../providers';
+import { IdentityClient } from '../../identity';
+import { PluginEndpointDiscovery } from '@backstage/backend-common';
+
+const Strategy = require('passport-strategy');
+// Using singleton keyStore instead of membership due to Passport wtf
+let keyStore: JWKS.KeyStore;
+let keyStoreUpdated: number;
+
+export class BackstageIdentityStrategy extends Strategy {
```

Comment:
Tbh I think it's best to make a cleaner implementation of this that doesn't have to deal with all the passport oddities and then use a much thinner wrapper if a passport strategy is needed. Looks good tho! 👍

Reply:
I switched back and forth a few times on this. There is an older commit with just the JWT verification for example. My last judgement was that Fredrik seemed to lean towards using available libraries to give adopters bigger flexibility which made passport a sexier choice. Might also be that my implementation is the odd part, and we simply document how to configure the available [JwtStrategy](http://www.passportjs.org/packages/passport-jwt/)? I opted this way to minimize adopter code, but didn't succeed to get it as pretty as I imagined. WDYT? Simple JWT verification, documenting JwtStrategy, rewriting this implementation or something else?